consensus/istanbul/backend/backend_test.go

```diff
@@ -311,42 +706,84 @@ func Test_GossipSubPeerTargets(t *testing.T) {
 	}
 	backend.core = istanbulCore.New(backend, backend.config)
 
+	// Test for blocks from 0 to maxBlockNum
 	for i := int64(0); i < maxBlockNum; i++ {
-		backend.currentView.Store(&istanbul.View{Sequence: big.NewInt(i), Round: big.NewInt(0)})
-		valSet.SetBlockNum(uint64(i))
-		valSet.CalcProposer(valSet.GetProposer().Address(), 0)
-		targets := backend.GossipSubPeer(common.BigToHash(big.NewInt(i)), valSet, nil)
+		// Test for round 0 to round 14
+		for round := int64(0); round < 15; round++ {
+			backend.currentView.Store(&istanbul.View{Sequence: big.NewInt(i), Round: big.NewInt(round)})
+			valSet.SetBlockNum(uint64(i))
+			valSet.CalcProposer(valSet.GetProposer().Address(), uint64(round))
 
-		// Test if the testing node is in the committee in specified blocks and target node list is returned
-		if (checkInCommitteeBlocks(i) && targets == nil) || (!checkInCommitteeBlocks(i) && targets != nil) {
-			t.Errorf("Wrong committee information. Block: %d, Nil targets: %v\n", i, targets == nil)
-		}
+			// Use block number as prevHash. In SubList() only left 15 bytes are being used.
+			hex := fmt.Sprintf("%015d000000000000000000000000000000000000000000000000000", i)
+			prevHash := common.HexToHash(hex)
+
+			// current and next committee
+			committees := make([][]istanbul.Validator, 2)
+
+			viewCurrent := backend.currentView.Load().(*istanbul.View)
+			committees[0] = valSet.SubList(prevHash, viewCurrent)
+
+			viewCurrent.Round = viewCurrent.Round.Add(viewCurrent.Round, common.Big1)
+			backend.currentView.Store(viewCurrent)
+
+			valSet.CalcProposer(valSet.GetProposer().Address(), uint64(round+1))
+			committees[1] = valSet.SubList(prevHash, viewCurrent)
+
+			// Reduce round by 1 to set round to the current round before calling GossipSubPeer
+			viewCurrent.Round = viewCurrent.Round.Sub(viewCurrent.Round, common.Big1)
+			valSet.CalcProposer(valSet.GetProposer().Address(), uint64(round))
+			backend.currentView.Store(viewCurrent)
 
-		if checkInCommitteeBlocks(i) {
-			// targets don't have testing node's address because a target is a receiving node
-			if len(targets) == len(council)-1 {
-				for _, x := range council {
-					// council[0] is testing node and it should be removed from targets
-					if x == council[0] {
-						continue
+			targets := backend.GossipSubPeer(prevHash, valSet, nil)
+
+			// Check if the testing node is in a committee
+			check1 := backend.checkInSubList(prevHash, valSet)
+			check2 := checkInCommitteeBlocks(i, round)
+
+			if check1 != check2 {
```

Comment:
Instead of using `check1` and `check2`, how about using variables with semantic names?

Reply:
@ehnuje Agreed. I'll change its names